src/library_pthread.js

```diff
@@ -55,6 +58,11 @@ var LibraryPThread = {
     // the reverse mapping, each worker has a `pthread_ptr` when its running a
     // pthread.
     pthreads: {},
+#if MAIN_MODULE
+    // Zombie threads are threads that have finished running but we not yet
+    // joined.
+    zombieThreads: {},
```

Comment:
Instead of having a new member `zombieThreads`, could we track a boolean in the existing `PThread.pthreads[pthread_ptr]` object? (e.g. `isDetached == false/true`?) Now to think of it, I think this info should already exist somewhere, since the system has to distinguish between detached, and unjoined nonterminated threads.

To me the term "zombie thread" means synonymous to "leaked thread", i.e. a thread that code failed to join with. We cannot know which threads actually are zombie threads until the program quits.

The accurate term would be "unjoined nondetached thread", but that is a bit of a mouthful. Well, 'unjoined' implies 'nondetached', so maybe these can be called just unjoined threads? Or (pendingJoin as a boolean field, if needed? Though I am surprised if the existing state doesn't have a way to distinguish these)